ks3-standard/src/commonTest/kotlin/io/ks3/standard/base64/ByteArrayAsBase64StringSerializerTest.kt

```diff
@@ -0,0 +1,36 @@
+package io.ks3.standard.base64
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.serialization.json.Json
+import kotlin.text.Charsets.UTF_8
+
+class ByteArrayAsBase64StringSerializerTest : FreeSpec(
```

Comment:
We have our work cut out when it comes to testing. Even with JSON you have to consider that the value could be 

* a map key
* a map value
* object key
* object value
* a collection value

And that's just JSON, other formats might have more 'positions' to consider.

But with regards to this test, because it's a string it should be okay. I'm brining it up because I'm thinking about testing all serializers in a generic way, like how KxS has some test utils for JSON https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonTestBase.kt

Reply:
Yeah, I also started to feel the need for some generalized test strategy. I'll give it some further thought and check out the link. I think a pattern will emerge after doing some more serializers and we could try to find a way to generalize it from there.

Reply:
With regards to position.. does the position really matter? I thought the serializer is applied in the same fashion regardless of position

Reply:
> With regards to position.. does the position really matter? I thought the serializer is applied in the same fashion regardless of position

Yeah apparently, that's the feedback I got during the BigDecimal PR https://github.com/Kotlin/kotlinx.serialization/pull/2041#discussion_r989143129

I think it's related to this issue https://github.com/Kotlin/kotlinx.serialization/issues/1966, which was some edgecase related to quoting unsigned numbers as map keys.